cl.cxx

```diff
@@ -0,0 +1,436 @@
+/**
+ * @file cl.cxx
+ * @author John Parent (john.parent@kitware.com)
+ * @brief A C++ wrapper file for the MSVC c and c++ compilers and linkers
+ *        created for the Spack package manager.
+ *
+ *        This file implements the functionality required to inject Spack's build logic
+ *        into the compiler/linker interface and drives the main entrypoint.
+ * @version 0.1
```

Comment:
IMO it's easier to track the version outside of this (e.g. using git tags).

Reply:
The Spack package will be tracked by the Git tags (or git releases if we so chose to cut those) but IMO versioning for software should be independent of its VCS.

Reply:
I don't work on any large C++ projects, but I think having a separate `version.hxx` file would be preferable, so we can update it more explicitly (I suppose as long as it lives in one place, then I don't mind version info in the files themselves, but I think making it clearer could be useful).

Reply:
Works for me!